sdk/nodejs/tests/runtime/langhost/run.spec.ts

```diff
@@ -1924,14 +1924,26 @@ function mockRun(
         runReq.setInfo(info);
 
         runReq.setDryrun(dryrun);
-        langHostClient.run(runReq, (err: Error, res: any) => {
-            if (err) {
-                reject(err);
-            } else {
-                // The response has a single field, the error, if any, that occurred (blank means success).
-                resolve([res.getError(), res.getBail()]);
+        // Sometimes it takes a little bit until the engine is ready to accept connections.  We'll
+        // retry a few times to make sure we don't fail spuriously.
+        for (let i = 0; i < 20; i++) {
+            let retry = false;
+            langHostClient.run(runReq, (err: Error, res: any) => {
```

Comment:
Pretty sure run is async so you actually need to do a chained callback inside run when handling the error.